sdk-lib/src/main/java/cash/z/ecc/android/sdk/internal/transaction/TransactionEncoderImpl.kt

```diff
@@ -47,25 +47,39 @@ internal class TransactionEncoderImpl(
      */
     override suspend fun createTransaction(
         usk: UnifiedSpendingKey,
+        account: Account,
         amount: Zatoshi,
         recipient: TransactionRecipient,
         memo: ByteArray?
     ): EncodedTransaction {
-        require(recipient is TransactionRecipient.Address)
+        require(recipient is TransactionRecipient.RecipientAddress)
 
-        val transactionId = createSpend(usk, amount, recipient.addressValue, memo)
+        val transactionId =
+            createSpend(
```

Comment:
It is actually `createSpend` that needs to change, and then that can propagate up.